archinstall/lib/disk/user_guides.py

```diff
@@ -122,8 +122,8 @@ def suggest_multi_disk_layout(block_devices, default_filesystem=None, advanced_o
 	layout[root_device.path]['partitions'].append({
 		# Boot
 		"type" : "primary",
-		"start" : "5MB",
-		"size" : "513MB",
+		"start" : "0",
```

Comment:
This will break GRUB, it needs somewhere between 2-5MB in order to install to the boot sector of a harddrive.
This picture taken from https://www.akadia.com/services/dual_boot.html describes it pretty nicely:
![dual_boot_disk_layout](https://user-images.githubusercontent.com/861439/145783284-fa09d8f0-d265-40cd-b169-5653a9ac6bea.gif)

Reply:
Is there a way to check for a BIOS within the user_guided.py script? I don't believe it needs the 512 MB boot partition like the UEFI does, and UEFI doesn't need the 5 MB even with GRUB, so there could be a part of the script that leaves 5 MB for GRUB on the BIOS and makes a 512 MB partition for the UEFI.

Reply:
Fixed, script now accounts for BIOS and UEFI setups and leaves 2 MB unallocated space as recommended by https://wiki.archlinux.org/title/GRUB#Master_Boot_Record_(MBR)_specific_instructions

Reply:
Yep, did some digging and I found out that hardware.py had that convenient function.

Before you made that post I used it in 01749ec, which added a small if statement that changes some variables if there is a UEFI in order to make the most optimal layout. Should I remove this?

Reply:
You can leave it in i think. I'll have a look at the overall PR and it it doesn't add too much complexity it should be fine :)